benchmarks/src/main/java/io/grpc/benchmarks/driver/LoadClient.java

```diff
@@ -0,0 +1,522 @@
+        * intervalHistogram.getTotalCount());
+    // TODO: No support for sum of squares
+
+    statsBuilder.setTimeElapsed(((double)(intervalHistogram.getEndTimeStamp()
+        - intervalHistogram.getStartTimeStamp())) / 1000.0);
+    if (osBean != null) {
+      // Report all the CPU time as user-time  (which is intentionally incorrect)
+      long nowCpu = osBean.getProcessCpuTime();
+      statsBuilder.setTimeUser(((double) nowCpu - lastMarkCpuTime) / 1000000000.0);
+      lastMarkCpuTime = nowCpu;
+    }
+    return statsBuilder.build();
+  }
+
+  /**
+   * Shutdown the scenario as cleanly as possible.
+   */
+  void shutdownNow() {
+    shutdown = true;
+    for (int i = 0; i < channels.length; i++) {
+      // Initiate channel shutdown
+      channels[i].shutdown();
+    }
+    for (int i = 0; i < channels.length; i++) {
+      try {
+        // Wait for channel termination
+        channels[i].awaitTermination(1, TimeUnit.SECONDS);
+      } catch (InterruptedException ie) {
+        channels[i].shutdownNow();
+      }
+    }
+    fixedThreadPool.shutdownNow();
+  }
+
+  /**
+   * Record the event elapsed time to the histogram and delay initiation of the next event based
+   * on the load distribution.
+   */
+  private void delay(long alreadyElapsed) {
+    long nextPermitted = distribution.sample();
+    recorder.recordValue(alreadyElapsed);
+    if (nextPermitted > alreadyElapsed) {
+      LockSupport.parkNanos(nextPermitted - alreadyElapsed);
+    }
+  }
+
+  /**
+   * Worker which executes blocking unary calls. Event timing is the duration between sending the
+   * request and receiving the response.
+   */
+  class BlockingUnaryWorker implements Runnable {
+    final BenchmarkServiceGrpc.BenchmarkServiceBlockingStub stub;
+
+    private BlockingUnaryWorker(BenchmarkServiceGrpc.BenchmarkServiceBlockingStub stub) {
+      this.stub = stub;
+    }
+
+    public void run() {
+      while (!shutdown) {
+        long now = System.nanoTime();
+        stub.unaryCall(simpleRequest);
+        delay(System.nanoTime() - now);
+      }
+    }
+  }
+
+  /**
+   * Worker which executes async unary calls. Event timing is the duration between sending the
+   * request and receiving the response.
+   */
+  class AsyncUnaryWorker implements Runnable {
+    final BenchmarkServiceGrpc.BenchmarkServiceStub stub;
+    final Semaphore maxOutstanding = new Semaphore(config.getOutstandingRpcsPerChannel());
+
+    AsyncUnaryWorker(BenchmarkServiceGrpc.BenchmarkServiceStub stub) {
+      this.stub = stub;
+    }
+
+    @Override
+    public void run() {
+      while (!shutdown) {
+        maxOutstanding.acquireUninterruptibly();
+        stub.unaryCall(simpleRequest, new StreamObserver<Messages.SimpleResponse>() {
+          long now = System.nanoTime();
+          @Override
+          public void onNext(Messages.SimpleResponse value) {
+
+          }
+
+          @Override
+          public void onError(Throwable t) {
+
```

Comment:
shouldn't `maxOutstanding` also be released here and the error should at least be logged?